src/Bladeburner/ui/ActionHeader.tsx

```diff
@@ -0,0 +1,55 @@
+import type { Bladeburner } from "../Bladeburner";
+import type { Action } from "../Types";
+
+import React from "react";
+import { Box, Typography } from "@mui/material";
+import { CopyableText } from "../../ui/React/CopyableText";
+import { createProgressBarText } from "../../utils/helpers/createProgressBarText";
+import { StartButton } from "./StartButton";
+import { StopButton } from "./StopButton";
+import { TeamSizeButton } from "./TeamSizeButton";
+
+import { formatNumberNoSuffix } from "../../ui/formatNumber";
+import { BlackOperation, Operation } from "../Actions";
+
+interface ActionHeaderProps {
+  bladeburner: Bladeburner;
+  action: Action;
+  rerender: () => void;
+}
+export function ActionHeader({ bladeburner, action, rerender }: ActionHeaderProps): React.ReactElement {
+  const isActive = action.name === bladeburner.action?.name;
+  const computedActionTimeCurrent = Math.min(
+    bladeburner.actionTimeCurrent + bladeburner.actionTimeOverflow,
+    bladeburner.actionTimeToComplete,
+  );
+  const allowTeam = action instanceof Operation || action instanceof BlackOperation;
+
+  if (isActive) {
+    return (
+      <>
+        <Box display="flex" flexDirection="row" alignItems="center">
+          <CopyableText value={action.name} />
+          <StopButton bladeburner={bladeburner} rerender={rerender} />
+        </Box>
+        <Typography>
+          (IN PROGRESS - {formatNumberNoSuffix(computedActionTimeCurrent, 0)} /{" "}
+          {formatNumberNoSuffix(bladeburner.actionTimeToComplete, 0)})
+        </Typography>
+        <Typography>
+          {createProgressBarText({
+            progress: computedActionTimeCurrent / bladeburner.actionTimeToComplete,
```

Comment:
There was a recent change to grafting and programs so they'd show ETA, does that make sense for BB as well?

Reply:
It's a good suggestion. I'll make a PR for it.